src/loops/simple_programs_with_repitition/easy/q7/MainTest.java

```diff
@@ -0,0 +1,60 @@
+package loops.simple_programs_with_repitition.easy.q7;
+
+import global.variables.*;
+import global.variables.clauses.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+
+public class MainTest extends BaseTest {
+    // Parsons
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "This is a sentence. This is another sentence.";
+
+        return new Clause[] {
+                new StringLiteral("Enter a sentence: "),
+                new NewLine(),
+                new StringLiteral("YoUr SeNtEnCe In AlTeRnAtInG cAsE iS: "),
+                new PlaceHolder()
+        };
+    }
+
+    public void runMain() {
+        AlTeRnAtInGcAsE.main(new String[0]);
+    }
+
+    public static String alternate(String sentence) {
+        String result = "";
+        int i = 0;
+        while (i < sentence.length()) {
+            if (i % 2 == 0)
+                result += Character.toUpperCase(sentence.charAt(i));
+            else
+                result += Character.toLowerCase(sentence.charAt(i));
+            i++;
+        }
+        return result;
+    }
+
+    static Stream<Arguments> inputProvider(){
+        return Stream.of(Arguments.of("This is just for testing. Do not copy!", alternate("This is just for testing. Do not copy!")), Arguments.of("This is second test", "ThIs iS SeCoNd tEsT"), Arguments.of("THIS IS ANOTHER HARDCODED TEST", "ThIs iS AnOtHeR HaRdCoDeD TeSt"), Arguments.of(" ", " ")); // Doesn't work with empty string
```

Comment:
Wait not like this, either hardcode the string as the second parameter of the `.of()` as a string (i.e. without the helper method) or use the helper method only in the `testWithInput()`, doing both is redundant since you're injecting a value using the helper method and then checking if it equals another value computed with the helper method. The problem being that if the helper method is ever wrong, the test will still pass since it only depends on itself